pages/maker/apps/[appId].tsx

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import { useRouter } from 'next/router';
+import { Flex, Box, Type } from 'blockstack-ui';
+import { connect, useSelector } from 'react-redux';
+import isNaN from 'lodash/isNaN';
+
+import {
+  selectMaker,
+  selectAppList,
+  selectCurrentApp
+} from '@stores/maker/selectors';
+import { fetchApps, selectAppAction } from '@stores/maker/actions';
+import { selectApiServer, selectUser } from '@stores/apps/selectors';
+import {
+  MakerContainer,
+  MakerContentBox,
+  MakerStickyStatusBox
+} from '@components/maker/styled';
+import { Page } from '@components/page';
+import { MakerNav } from '@components/maker/nav/maker-nav';
+import Head from '@containers/head';
+import Maker from '@components/maker';
+import UserStore from '@stores/user';
+
+const mapStateToProps = (state: any) => ({
+  apiServer: selectApiServer(state),
+  appList: selectAppList(state)
+});
+
+const handleChangingApp = (event: any, fn: any) => (dispatch: any) => {
+  event.persist();
+  const id = event.target.value;
+  dispatch(selectAppAction(id));
+  fn(id);
+};
+
+const getAppId = (params: any) => {
+  if (!params) return undefined;
+  const id = parseInt(params.appId, 10);
+  if (isNaN(id)) return undefined;
+  return id;
+};
+
+const LoadingPage = ({ message = 'Loading...' }) => (
+  <Page innerPadding={[0]} wrap>
+    <Flex>
+      <Box width={1}>
+        <Type fontSize={5} my={7} textAlign="center">
+          {message}
+        </Type>
+      </Box>
+    </Flex>
+  </Page>
+);
+
+const MakerPortal = connect()(({ appList, apiServer, dispatch }: any) => {
+  const router = useRouter();
+
+  const { user, maker, selectedApp } = useSelector(state => ({
+    user: selectUser(state),
+    maker: selectMaker(state),
+    selectedApp: selectCurrentApp(state)
+  }));
+
+  const updateMakerRoute = (id: number) => router.push(`/maker/apps/${id}`);
+
+  if (maker.loading || !selectedApp) return <LoadingPage />;
+
+  const subNav = (
+    <MakerNav
+      apps={appList}
+      userId="kyranjamie.id"
+      selectedAppId={selectedApp.id}
+      handleSignOut={() => {
+        dispatch(UserStore.actions.signOut());
+        localStorage.clear();
+        window.location.href = '/';
+      }}
+      onChange={e => handleChangingApp(e, updateMakerRoute)(dispatch)}
+    />
+  );
+
+  return (
+    <Page innerPadding={[0]} subNav={subNav} wrap>
+      <Head title={selectedApp.name} />
+      <MakerContainer>
+        <Type fontSize={3} fontWeight={500} mx={[4, 6]} py={6} px={[20, 0]}>
+          {selectedApp.name}
+        </Type>
+        <Flex flexDirection={['column', 'column', 'row-reverse']}>
+          <MakerStickyStatusBox>
+            <Maker.Status app={selectedApp} />
+          </MakerStickyStatusBox>
+          <Box>
+            <MakerContentBox>
+              <Maker.PaymentDetails
+                app={selectedApp}
```

Comment:
Nit: I know you're just re-using the code I originally wrote here, but it would probably be cleaner to just select these props from redux inside of the component.